source/grid/manifold_lib.cc

```diff
@@ -178,6 +152,78 @@ SphericalManifold<dim,spacedim>::push_forward_gradient(const Point<spacedim> &sp
 }
 
 // ============================================================
+// SphericalManifold
+// ============================================================
+
+template <int dim, int spacedim>
+SphericalManifold<dim,spacedim>::SphericalManifold(const Point<spacedim> center):
+  center(center)
+{}
+
+template <int dim, int spacedim>
+Point<spacedim>
+SphericalManifold<dim,spacedim>::
+get_new_point (const Point<spacedim> &P1, const Point<spacedim> &P2, double w) const
+{
+  if ( P1 == P2 )
+    return P1;
+
+  if (dim == spacedim)
+    {
+      return Point<spacedim>((1-w) * P1 + w * P2);
+    }
+  else
+    {
+      Assert(w >=0.0 && w <= 1.0,
+             ExcMessage("w should be in the range [0.0,1.0]."));
+      Assert((center-P1).norm() == (center-P2).norm(),
+             ExcMessage("P1 and P2 should have the same norm."));
+
+      Tensor<1,spacedim> V1 = P1 - center;
+      Tensor<1,spacedim> V2 = P2 - center;
+      double R = V1.norm();
+      double gamma = 2 * std::asin( P2.distance(P1) / (2*R) );
+      double sigma = w * gamma;
+      Tensor<1,spacedim> n = V2/R - ((V2/R)*(V1/R))*(V1/R);
+      n = n/n.norm();
+      Tensor<1,spacedim> V = std::cos(sigma) * V1 + std::sin(sigma) * R * n;
+      return Point<spacedim>(center + V/V.norm());
+    }
+}
```

Comment:
This implementation drastically changes the behaviour of the current SphericalManifold. I don't think this is a good idea. The implementation should be the same in codim 0 and 1. 

It should coincide with geodesics on codimension 1 surfaces, and with arcs on the plane identified by P1, P2 and the center, where the radius changes linearly between $P1$ and $P2$, and so does theta.

You are representing a FlatManifold in codimension 0 and the old HyperBallBoundary in codimension 1, making this class somewhat less useful.

The difference between this and PolarManifold should only be that this one does *not* break at the poles. From the way you implemented this, applying SphericalManifold or not to a dim=spacedim triangulation does not make any difference...